### PR TITLE
Add Gmail thread ID to queries, harden cache validation, and clean up API layer

### PR DESCRIPTION
**Pull request by robelkin**
## Summary

This started as adding Gmail thread IDs to MCP responses but grew to include cache validation hardening and API cleanup that came up along the way.

### 1. Gmail Thread ID (`source_conversation_id`) in query layer
- Add `SourceConversationID` field to `MessageSummary` and `MessageDetail` models
- DuckDB engine joins `conversations` parquet table; SQLite engine joins `conversations` table
- Export `conversations` table to Parquet in `build-cache` (including Windows CSV fallback path)
- MCP tools (`list_messages`, `get_message`, `search_messages`) now return `source_conversation_id`

### 2. Cache validation hardening
- Shared `RequiredParquetDirs` list used by cache builder, TUI, and MCP startup
- `HasCompleteParquetData()` checks all required tables exist (not just messages)
- TUI and MCP use `HasCompleteParquetData` instead of `HasParquetData` — prevents DuckDB errors when tables like `conversations` are missing
- `missingRequiredParquet()` detects partial/broken caches and triggers full rebuild (handles hive-partitioned messages layout)
- `buildCacheMu` mutex prevents concurrent cache writes from parallel syncs
- `maxID > 0` gate avoids infinite rebuild loops for zero-message accounts
- +482 lines of test coverage for cache edge cases

### 3. Remove dead config field and eliminate DTO duplication
- Remove unused `server.mcp_enabled` config field (MCP is always available via `msgvault mcp`)
- Replace duplicate `APIMessage`, `APIAttachment`, `StoreStats`, `AccountStatus` structs in `internal/api` with type aliases to `store`/`scheduler` types
- Collapse ~80 lines of field-by-field adapter boilerplate in `serve.go` to direct pass-throughs
- Add backward-compat test for old configs that still contain `mcp_enabled`

## Test plan
- [x] All existing tests pass
- [x] New cache tests cover: backfill missing conversations, backfill missing messages, zero-message skip, incremental no-duplicates, hive-partitioned detection
- [x] MCP tests verify `source_conversation_id` in search/get/list responses
- [x] Config backward-compat test for deprecated `mcp_enabled`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by wesm** [comment]
Sorry Claude made a mess, I think I fixed it now

File: internal/api/server.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+		next.ServeHTTP(ww, r)
+	})
+}
+
+// authMiddleware validates the API key.
+func (s *Server) authMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Skip auth if no API key configured
+		if s.cfg.Server.APIKey == "" {
```
[the remark this answers is not in the document]
Comment (hughdbrown) [reply internal/api/server.go:192]: > However, the code strips the 'Bearer ' prefix with a simple string slice (line 203) before comparison. This prefix stripping happens in non-constant time.

Is the claim that the comparison of `s.cfg.Server.APIKey` to the empty string happens in a variable amount of time?

**Comment by wesm** [comment]
@hughdbrown if you have any concerns about changes in this PR (and [d24c017](https://github.com/wesm/msgvault/commit/d24c017b9662a402152e13327655d671625afc60)) please go ahead and open up a follow up PR!
